services/impl/src/main/java/org/exoplatform/automatic/translation/impl/AutomaticTranslationServiceImpl.java

```diff
@@ -26,84 +27,111 @@
+    String activeConnector = getActiveConnector();
+    AutomaticTranslationConfiguration configuration = new AutomaticTranslationConfiguration(new ArrayList<>(), activeConnector);
+    getConnectors().forEach((name, connector) -> {
+      if (name.equals(activeConnector)) {
+        configuration.setActiveApiKey(connector.getApiKey());
+        configuration.addConnector(name, connector.getDescription(), connector.getApiKey());
+      } else {
+        configuration.addConnector(name, connector.getDescription(), null);
+      }
+    });
+    return configuration;
+  }
+
   @Override
   public String getActiveConnector() {
     SettingValue<?> value = settingService.get(Context.GLOBAL, Scope.GLOBAL, AUTOMATIC_TRANSLATION_ACTIVE_CONNECTOR);
 
-
-    return value != null && value.getValue()!=null && !value.getValue().toString().isEmpty() && isActiveFeature() ?
-           value.getValue().toString() : null;
+    return value != null && value.getValue() != null && !value.getValue().toString().isEmpty()
+        && isActiveFeature() ? value.getValue().toString() : null;
   }
 
   private boolean isActiveFeature() {
-    String enable = System.getProperty("exo.feature."+FEATURE_NAME+".enabled");
-    return (enable!=null && !enable.isBlank() && enable.equals("true"));
-    //after feature finished, return this :
-    //return exoFeatureService.isActiveFeature(FEATURE_NAME);
+    return exoFeatureService.isActiveFeature(FEATURE_NAME);
   }
 
   @Override
-  public boolean setActiveConnector(String name) {
+  public void setActiveConnector(String name) {
     if (!isActiveFeature()) {
       LOG.error("Try to change automatic translation connector but feature is not active");
-      return false;
+      throw new RuntimeException("Unable to change automatic translation connector as feature is not active");
     }
-    //should return false if connector name do not exists
-    //but not if the name is empty (correspond to unset case)
+    // should return false if connector name do not exists
+    // but not if the name is empty (correspond to unset case)
     if (name == null) {
       name = "";
     }
     String finalName = name;
-    if (finalName.isBlank() ||
-        translationConnectors.keySet().stream().anyMatch(connectorName -> connectorName.equals(finalName))) {
+    if (finalName.isBlank()
+        || translationConnectors.keySet().stream().anyMatch(connectorName -> connectorName.equals(finalName))) {
       settingService.set(Context.GLOBAL, Scope.GLOBAL, AUTOMATIC_TRANSLATION_ACTIVE_CONNECTOR, new SettingValue<>(name));
-      return true;
     } else {
-      LOG.error("Try to change automatic translation connector with a non existing connector name : {}",name);
-      return false;
+      LOG.error("Try to change automatic translation connector with a non existing connector name : {}", name);
+      throw new RuntimeException("Unable to change automatic translation connector as provided connector name do not exists");
```

Comment:
please avoid logging and throwing at the same time the error. Either log or throw the error. (It will cause a double trace for the same error)